docs/checkpoint-settings.md

```diff
@@ -0,0 +1,117 @@
+# Checkpoint Settings
+
+For materialized views, checkpoint settings control how and where checkpoints are created, as well as whether they are incremental or asynchronous. This can be specified in the `checkpoint_settings` section of the materialized view settings.
+
+```sql
+CREATE MATERIALIZED VIEW mv AS
+SELECT key, count() FROM test group by key
+SETTINGS default_hash_table='hybrid', default_hash_join='hybrid',
+checkpoint_settings = 'incremental=true;interval=5';
+```
+
+## checkpoint_settings
+You can set key-value pairs in `checkpoint_settings`.
+
+### type
+
+**Definition**: Defines which checkpoint type to use.
+
+**Possible Values**:
+
+- `auto`  **(default)** - Automatically determine whether to use `file` or `rocks` checkpoint based on the query’s state type.
+- `file` - Native file format
+- `rocks` - RocksDB format
```

Comment:
'rocks' currently does not support manual setting, it is automatically determined by the system internally

Reply:
I also had questions when I saw the original list. So if the system figured the stream can use `rocks`, can the user set `file` explicitly?

Reply:
yes , we can specify all state in file format, but the rocks format depends on whether the query state supports it.